### PR TITLE
Adding User routes

### LINKED ISSUE #23: Update a user
Handle a PUT request on the user route.
The endpoint should accept a name, email an password for an update.

### LINKED ISSUE #24: Delete a user
Handle a DELETE request on the user route.
The request should delete and logout the current user.

### PR DESCRIPTION
**Pull request by abelRoland**
This closes #23  and this closes #24

### CODE REVIEW

**Review by Eltayeb-Elgaali — COMMENTED** [review]
It seems good to me.
the only thing is the route that should be may be changed.

File: api/routes/user.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    User
+        .deleteOne({ _id: id })
+        .then(result => {
+            res.send(result)
+        })
+        .catch(err => res.status(400).send(err))
+    ;
+}
+
+router.put(
```
Comment (Toinne) [inline api/routes/user.js]: Requirement: make the route `/user`

**Review by Toinne — CHANGES_REQUESTED** [review]

**Review by Sayed94h — APPROVED** [review]
Now it is how we discussed in our previous meeting so I am going to approve it.

**Review by HazemBittar — APPROVED** [review]

File: api/routes/user.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        .then(result => {
+            return res.status(200);
+        })
+        .catch(err => {
+            return res.status(400).send(err)
+        })
+    ;
+}
+
+app.put('/users/:id/update', auth, updateUser);
```
Comment (Toinne) [inline api/routes/user.js:57]: @abelRoland it is very important to authenticate these routes with auth!! If not this endpoint is public for the entire internet.

File: api/routes/user.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+const User = require('../models/User');
+const express = require('express');
+const { auth } = require('../middleware');
+
+const app = express();
+
+const updateUser = (req, res) => {
+    const { id } = req.params;
+
+    if (req.user._id != id) {
```
Comment (Toinne) [inline api/routes/user.js:10]: @abelRoland you only want to allow updating your own user account. If you don't add this anyone on the internet can update the details of the user including password.

File: api/server.js [hunk 4]
```diff
@@ -14,5 +14,6 @@ app.use(express.static(path.join(__dirname, 'public')));
 
 app.use('/', require('./routes/home.js'));
 app.use('/', require('./routes/login.js'));
+app.use('/', require('./routes/user.js'));
```
Comment (Toinne) [inline api/server.js:18]: This should have been added @abelRoland

**Pull request merged**